### ISSUE TITLE
audio: expose the streaming session publicly — push PCM while the user speaks, keep only the 202 ms tail

### ISSUE THREAD

**Issue by geisten** [geisten/geistlib#256]
The engine already encodes audio incrementally while PCM arrives: the Phase-8b worker (`GEIST_AUDIO_STREAM=1`) runs 48-mel-frame Conformer batches during capture, bit-identical to the monolithic path (`test_audio_stream_live_parity_int`), leaving a **202 ms** post-utterance tail on the Pi 5 vs 3.6 s monolithic (`benchmark/results/PI5-audio.md`, #235).

But that machinery lives in the ARCH-internal `audio_encoder_push_pcm` / `pull_softtokens` API. The public surface (`geist_session_attach_audio`) is utterance-level: a caller like `examples/push_to_talk.c` must buffer the whole utterance and then pay the full encode — 2.8 s attach for a 2.7 s clip on the Pi instead of ~0.2 s. Real-time voice applications built on the *public* API cannot reach the latency the engine already delivers internally.

## Proposed shape (`@stability EXPERIMENTAL`, geist_util.h)

```c
/* Begin a streaming audio turn. The encoder consumes PCM as it arrives;
 * soft tokens are injected into the session on end(). */
enum geist_status geist_session_audio_begin(struct geist_session *s);

/* Append 16 kHz mono s16 PCM. Cheap (memcpy + mel increment + worker
 * kick); safe to call from a capture thread. */
enum geist_status geist_session_audio_push(struct geist_session *s,
                                           size_t n, const int16_t pcm[static n]);

/* End of utterance: finish the tail, inject the soft tokens, return.
 * After this the session is exactly as if attach_audio had been called
 * with the concatenated PCM. */
enum geist_status geist_session_audio_end(struct geist_session *s);
```

Semantics anchor: `begin → push* → end` ≡ `attach_audio(concat(pcm))` — that equivalence is the testable contract. Errors (no tower, wrong rate, >30 s buffer) surface with the same codes attach_audio uses; `geist_model_modalities()` continues to answer capability up front.

## Acceptance

- [ ] **Equivalence test**: a new int test drives `begin/push/end` with 20 ms chunks and asserts the session state matches an `attach_audio` session — same injected token count and (via `peek_logits`) identical next-token logits. Runs mandatorily in the `audio-smoke` job.
- [ ] **Latency**: on the reference Pi 5, `end()` on a 10 s utterance pushed at real-time rate returns in ≤ 300 ms (the measured worker tail is 202 ms) — recorded in `PI5-audio.md`, not asserted on shared CI runners.
- [ ] **Thread-safety**: the capture-thread pattern (audio thread pushes, inference thread ends/decodes) runs under the TSan CI leg without findings.
- [ ] **First consumer**: `examples/push_to_talk.c` switches from buffer-then-attach to push-per-VAD-frame and prints the measured tail — the example is the documentation.
- [ ] EXPERIMENTAL tag + CHANGELOG; not in the API contract until an out-of-tree consumer depends on it (#225 bar).
